lib/sepa.test.js

```diff
@@ -30,3 +36,135 @@ describe('IBAN tests',
       expect(SEPA.validateIBAN('DE1Zsantander')).toBe(false);
     });
   });
+
+describe('xml generation for transfer documents', () => {
+  const PAIN_FOR_TRANSFERS = 'pain.001.003.03';
+
+  function validTransferDocument({debtorId=A_VALID_CREDITOR_ID, debtorName='debtor-name'}) {
+    const doc = new SEPA.Document(PAIN_FOR_TRANSFERS);
+    doc.grpHdr.created = new Date();
+
+    const info = doc.createPaymentInfo();
+    info.collectionDate = new Date();
+    info.debtorIBAN = A_VALID_IBAN;
+    info.debtorName = debtorName;
+    info.debtorId = debtorId;
+    info.requestedExecutionDate = new Date();
+    doc.addPaymentInfo(info);
+
+    const tx = info.createTransaction();
+    tx.creditorName = 'creditor-name';
+    tx.creditorIBAN = A_VALID_IBAN;
+    tx.amount = 1.0;
+    tx.mandateSignatureDate = new Date();
+    info.addTransaction(tx);
+    return doc;
+  }
+
+  test('debtor id is optional for transfer documents', () => {
+    // GIVEN
+    const doc = validTransferDocument({debtorId: null});
+    // WHEN
+    const dom = doc.toXML();
+    // THEN
+    const select = xpath.useNamespaces({
+      p: `urn:iso:std:iso:20022:tech:xsd:${PAIN_FOR_TRANSFERS}`,
+    });
+    const debtorId = select('/p:Document/p:CstmrCdtTrfInitn/p:PmtInf/p:Dbtr/p:Id', dom, true);
+    expect(debtorId).toBeUndefined();
+  });
+
+  test('debtor name and id are included in transfer documents when they are set', () => {
+    // GIVEN
+    const doc = validTransferDocument({debtorId: A_VALID_CREDITOR_ID, debtorName: 'debtor-name'});
```

Comment:
This case for example seems to be using the defaults completely, so with case (2) you'd keep it as is, with case (1) you can use `validTransferDocument()`

Reply:
I have updated the PR by using  values  for the overrides that are different from the defaults. I hope that this a good compromise between readability of the expectations and avoiding confusion as to why the defaults were passed into the method.
Thanks for the feedback.

Reply:
Sure thing. Let's go with what you have, this isn't a big deal :) I don't think we need a major compromise, more the consistency. We can be explicit, though then we don't need default parameters since we'll always be passing them. I'll accept and merge this one in a moment.